packages/ripple-keypairs/src/index.ts

```diff
@@ -38,66 +45,79 @@ const secp256k1 = {
     privateKey: string
     publicKey: string
   } {
-    const prefix = '00'
-
+    const derived = derivePrivateKey(entropy, options)
     const privateKey =
-      prefix + derivePrivateKey(entropy, options).toString(16, 64).toUpperCase()
+      SECP256K1_PREFIX + bytesToHex(numberToBytesBE(derived, 32))
```

Comment:
This line seems to be using different bytes as far as I can understand? 

Previously the privateKey was only using 16-64, but I'm unclear how `numberToBytesBE` with a parameter of 32 can be extracting the same data. (Is the return type of `derivePrivateKey` significantly different now?)

Reply:
The toString() method there (which encodes a big number in big-endian) is is accepting 2 arguments. The first, with `16` as parameter, is the base to encode in, where base 16 is equivalent to hexadecimal. `64` is the number of hexadecimal characters to encode to, relevant if the number is small and the encoding needs to be zero-padded to a consistent width. 

The toString() params:

* 16: base of the encoding (i.e. hexadecimal, with one nibble per character)
* 64: length of encoded number, zero padded where necessary

When working with a Uint8Array, we are working with bytes, not nibbles, so 32 is appropriate. numberToBytesBE likewise zero-pads when necessary.

Reply:
Ah, that makes sense. I misinterpreted `toString` as operating like `substring`. Thanks for the detailed explanation :)